[tensorexpr] Set device for cuda-codegen if new device is not prior device

protonu opened the pull request.

Commit:
set device for cuda-codegen if new device is not prior device

Commit:
set device for cuda-codegen if new device is not prior device

zheng-xq commented on torch/csrc/jit/tensorexpr/cuda_codegen.cpp:
```diff
@@ -620,11 +620,18 @@ void CudaCodeGen::CompileToNVRTC(
+  }
   // cudaSetDevice does not have to really change the underlying device if it
   // doesn't have to, so calling cudaFree to force that change
   CudaSetContext(pctx);
```
Why keeping "CudaSetContext(pctx);", and the definition of "CudaSetContext"?

protonu replied:
Thanks for catching this. It must have crept in when I was moving to master. I am removing this.

Commit:
avoid redundant calls to cudaFree

Commit:
avoid redundant calls to cudaFree

Commit:
use nullptr to address clang-tidy

Commit:
use nullptr to address clang-tidy

Commit:
use nullptr to address clang-tidy

Commit:
use nullptr to address clang-tidy

Commit:
Updating submodules

Summary:
GitHub commits:

https://github.com/facebook/fbthrift/commit/0bd7af23b3cd69509e58fbd4b39a12bee0e426c1
https://github.com/facebook/folly/commit/4d2f1db9636d78e53c6ec9cfaa235870163c89af
https://github.com/facebook/rocksdb/commit/d300d10962ac952e0648ecc454b433ff29d6e96f
https://github.com/facebook/wangle/commit/9eafc02456b7d6f98f684fd470ed35374d978a38
https://github.com/facebookincubator/fizz/commit/a2f301b725fbb86ccd2ba44f9c99646a990dd35f
https://github.com/facebookincubator/katran/commit/b0395f0b0e70e63031a3bc9bd73b6b414b544592
https://github.com/facebookincubator/mvfst/commit/dcb403515c5b0ed5de71f0377b29da744c93b7b9
https://github.com/facebookresearch/pytorch-biggraph/commit/74a3c0ae53a750c316d3b4ab28aedf8daf258f1b
https://github.com/pytorch/fbgemm/commit/60b734a667576b659a717d5dbae97e1599c1cb2e

Test Plan: n/a

Reviewed By: wittgenst

fbshipit-source-id: 6fb6754598b16939b1b8acd7ea7d022dd7ee473c

peterbell10 committed:
FAQ: Add note about recovering from OOM (#35214)

Summary:
Closes https://github.com/pytorch/pytorch/issues/18853

This documents the workaround needed to solve the issues in https://github.com/pytorch/pytorch/issues/18853
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35214

Differential Revision: D20604877

Pulled By: ezyang

fbshipit-source-id: 71ed13cfa567d8e88fa9f18180a171cd174fb528

lly-zero-one committed:
Add the more fusion (conv3d and batchnorm)support in pytorch quantization flow (#33540)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/33540

Differential Revision: D19994498

Pulled By: lly-zero-one

fbshipit-source-id: e5e13eab6924bd2ce1b57b16b672844b8b9638f5

Commit:
[PyTorch-RPC] In process_group_agent, avoid read-after-free (#35252)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35252

The torch::from_blob() syntax without a deleter syntax is relatively
dangerous and explicitly assumes that the caller will correctly persist
the tensor bits for as long as necessary.

We were at one point correctly persisting the send tensor bits in
process_group_agent, but with the early-return codepaths are not
doing so any longer.

This change switches to a more robust approach where we instead just use
the torch::from_blob-with-deleter syntax, and use std::move to avoid
a copy. There's an extra malloc, but that's effectively free compared with
the rest of the work involved here. And it means we don't have to worry
about the Tensor memory vanishing from underneath the send anymore.

The initial motivation here was dist_autograd_node_failure flakiness.

While the motivating case is handleSend(), we also fix handlePendingMessage().
ghstack-source-id: 100704883

Test Plan:
existing test coverage, e.g.
   buck test mode/dev-nosan caffe2/torch/fb/distributed/thriftRpcBackend/test:ProcessGroupAgentTest

Differential Revision: D20607028

fbshipit-source-id: cf9966c5aa9472830cfefaf7fc2f92af9b52630d

jerryzh168 committed:
[quant][graphmode] Replicate quantize node for prim::If (#34804)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34804

We want to replicate the quantize node for return values in blocks of prim::If
in order to create the quantization patterns.

Test Plan:
python test/test_jit.py

Imported from OSS

Differential Revision: D20524453

fbshipit-source-id: 2268ac555f646158f4e1ffc98ccc8101d7504194

jspark1105 committed:
[caffe2] merge internal (RowWise)SparseAdagrad into open source version (#35090)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35090

As a preparation to open source fp16 + stochastic rounding SparseAdagrad and fused SparseAdagrad

Other minor changes:
* Removed template parameters T that are not actually used
* Removed unnecessary anonymous namespaces used in header files

Test Plan: CI

Reviewed By: jianyuh

Differential Revision: D20552770

fbshipit-source-id: 224fdca15ea786620ce88e33cbcbf97661423538

wanchaol committed:
[autograd] allow PyNode to persist error message (#34845)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34845

This PR allows PyNode to persist the error message so that any pure C++
thread that runs autograd with custom Python autograd function can successfully
catpure the error message without maintaining a initial PyThreadState.

Test Plan: Imported from OSS

Differential Revision: D20480685

Pulled By: wanchaol

fbshipit-source-id: 0488ea5a4df9a33b53ac5d0d59000c41ab6cb748

malfet committed:
Fix `JitTest.ADFormulas` intermittent failures (#35196)

Summary:
Clamp input tensor values to [3, 3] to limit how small `tanh` gradint can get
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35196

Test Plan: CI + `bin/test_jit --gtest_filter=JitTest.ADFormulas --gtest_repeat=60000 --gtest_break_on_failure`

Differential Revision: D20611256

Pulled By: malfet

fbshipit-source-id: 8640faa5d8567d6c6df8cc5df80c2e65407116eb

jerryzh168 committed:
[quant][graphmode] quantization support for aten::chunk (#34806)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/34806

Test Plan:
python test/test_jit.py

Imported from OSS

Differential Revision: D20524454

fbshipit-source-id: 92ac9bc251581e963258cb90dc3de73f8508c822

jerryzh168 committed:
[quant][graphmode] quantization support for view, transpose, contiguos (#34854)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/34854

Test Plan:
python test/test_jit.py

Imported from OSS

Differential Revision: D20524456

fbshipit-source-id: e6e8fc3db6cccbd32c210d04f921274d81996fe2

jerryzh168 committed:
[quant][graphmode] quantization support for prim::ListUnpack (#34807)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/34807

Test Plan:
python test/test_jit.py
in https://github.com/pytorch/pytorch/pull/34806

Imported from OSS

Differential Revision: D20524452

fbshipit-source-id: 31956894d6be58b6ba96b02d338dd1fd802aeefc

jerryzh168 committed:
[quant][graphmode] Add prim::ListConstruct to general op handling (#34345)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34345

prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Imported from OSS

Differential Revision: D20524455

fbshipit-source-id: b5f8e0c8776d48d588aeba6735de06dcd308560e

mruberry committed:
Extends true_divide to be a method (#34794)

Summary:
Per title. See related https://github.com/pytorch/pytorch/pull/34570.

In PyTorch 1.7 the plan is for torch.div and Python's division operator to perform "true" division, like Python 3, JAX, and NumPy. To facilitate this change, this PR expands true_divide to be a method so it can cover all of torch.div's use cases.

New true_divide tests are added to test_torch.py, test_type_promotion.py, and test_sparse.py.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34794

Differential Revision: D20545507

Pulled By: mruberry

fbshipit-source-id: 55286f819716c8823d1930441a69008560ac2bd5

Commit:
Revert D20539336: [JIT] add IR complexity tests

Test Plan: revert-hammer

Differential Revision:
D20539336

Original commit changeset: 14ac00a7b2b0

fbshipit-source-id: 1a51b461e88720599faf04dd3ca443d87f4de66d

Commit:
[JIT] BC shim for TorchBind classes (#35240)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35240

This makes it so that if we have an old serialized TorchBind class, we don't try to load it in and instead rely on the ClassType that's in memory.
ghstack-source-id: 100703946

Test Plan: buck test mode/dev-nosan //caffe2/torch/fb/predictor/model_repo/tests:ai_infra_representative_model_shard_6_test -- 'RepresentativeModelTest\/ShardedRepresentativeModelTest\.RunModel\/0'

Reviewed By: zdevito

Differential Revision: D20605681

fbshipit-source-id: 5403f68937f822914c701d9c80573f0b4a93e83b

Commit:
[reapply][JIT] Namespaces for TorchBind (#35254)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35254

Reapply D20541090 with some BC fixes
ghstack-source-id: 100733987

Test Plan: buck test mode/dev-nosan //caffe2/torch/fb/predictor/model_repo/tests:ai_infra_representative_model_shard_6_test -- 'RepresentativeModelTest\/ShardedRepresentativeModelTest\.RunModel\/0'

Reviewed By: zdevito

Differential Revision: D20607111

fbshipit-source-id: 80f148d860571208c93e9308128cd480ff089f74

peterjc123 committed:
Load torch_global_deps for Windows (#35177)

Summary:
Fixes https://discuss.pytorch.org/t/torch-cat-runtimeerror-error-in-loadlibrarya/71188/8.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35177

Differential Revision: D20604654

Pulled By: ezyang

fbshipit-source-id: 263eb401300812fd336ff820c53b543342dca95e

peterbell10 committed:
Fix input overwriting in irfft (#35219)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/34551
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35219

Differential Revision: D20605330

Pulled By: ezyang

fbshipit-source-id: a62f1685779bb05c3682255bb3a3f6f9ec35814f

anjali411 committed:
Gradcheck for complex (#35238)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/35238

Differential Revision: D20607581

Pulled By: anjali411

fbshipit-source-id: 2caf78314a87461b255fd65c7f71c72e152b5161

xuhdev committed:
Add cpu_serial_kernel_vec (#34553)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34553

This allows vectorized looping in a serial iteration over
TensorIterator.

Test Plan: Imported from OSS

Differential Revision: D20604238

Pulled By: ezyang

fbshipit-source-id: 61c451dac91d47cde7e1a937b271ab78c79e05d3

definitelynotmcarilli committed:
[RELAND2] Eager autocasting, out-of-place ops only (with MSVC 2017 fix) (#35102)

Summary:
This is the second reland attempt for https://github.com/pytorch/pytorch/pull/32140.

The first reland attempt https://github.com/pytorch/pytorch/pull/35011 failed due a [small incompatible change](https://github.com/pytorch/pytorch/pull/35011#issuecomment-601754216) in recent master (`skipIfRocm` was removed from `test_data_parallel.py`).

The present PR restores skipIfRocm.

Description from first reland attempt https://github.com/pytorch/pytorch/pull/35011:

> https://github.com/pytorch/pytorch/pull/32140 was approved and merged, but [reverted](https://github.com/pytorch/pytorch/commit/d0577e19f09a32a68f0f3faed635dec72971b019) because it broke builds with versions of Visual Studio older than 15.8 that were not represented in public CI.  The build failures were caused by a [known VS bug](https://developercommunity.visualstudio.com/content/problem/27729/allow-function-with-internal-linkage-as-template-n.html), fixed in versions 15.8 and newer.
>
> The present PR reverts the revert (restoring https://github.com/pytorch/pytorch/pull/32140 's diffs) and adds a workaround to enable compilation with VS < 15.8.  The workaround isn't pretty, but it's guarded by macros such that it's only used when compiling with VS < 15.8.  All other builds compile with the same code/control flow as was merged in https://github.com/pytorch/pytorch/pull/32140.
>
> Original description of https://github.com/pytorch/pytorch/pull/32140:
> > Initial integration of eager autocasting, supporting out-of-place ops only for easier review.
> Relevant issue/RFC: https://github.com/pytorch/pytorch/issues/25081
>
> > In-place ops and ops with user-supplied out=... can certainly be supported as well (my initial WIP https://github.com/pytorch/pytorch/issues/29552 handled many) but require substantially more complex special casing in the autocasting backend and tests. Support for these ops (much of which has already been written) will be broken into later PRs.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35102

Differential Revision: D20596918

Pulled By: ezyang

fbshipit-source-id: 60caa279bb0ce4a9bb0b28c1d585d42cf1cc7e50

xuhdev committed:
Replace all uses of AT_INDEX_ERROR with TORCH_CHECK_INDEX (#35050)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/35050

Differential Revision: D20550978

Pulled By: ezyang

fbshipit-source-id: df7c0730f27d2986601b1dee17e41957be2956de

xush6528 committed:
Add python excepiton handling catch block to resolve deadlock (#35283)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35283

https://github.com/pytorch/pytorch/issues/34260

Deadlock on destructing py::error_already_set.

There are request callback impls in Python, where Python exceptions could be thrown. For releasing Python exception py::objects, GIL must be held.

Test Plan:
```
buck test mode/dev-nosan //caffe2/test/distributed/rpc/jit:rpc_fork

buck build mode/dev-nosan //caffe2/test/distributed/rpc/jit:rpc_fork && \
buck-out/gen/caffe2/test/distributed/rpc/jit/rpc_fork\#binary.par \
-r test_torchscript_functions_not_supported
```

Differential Revision: D7753253

fbshipit-source-id: 4bfaaaf027e4254f5e3fedaca80228c8b4282e39

pritamdamania committed:
[Autograd Testing] Add a test where child reentrant task fails. (#35223)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35223

Adding tests as part of
https://github.com/pytorch/pytorch/issues/34367.

This test covers
"Mixed with errors" ->
"Reentrant on same device" ->
"Make child error before parent finishes"
ghstack-source-id: 100725947

Test Plan: waitforbuildbot

Differential Revision: D20603127

fbshipit-source-id: 08484b0a98053491459e076bdd23caf042c47150

supriyar committed:
[quant][graphmode] Add observers for dynamic quant (#35121)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35121

For dynamic quantization we insert observers at the input to mimic the quatization of activations that happens in the operator
Observer for weight is inserted similar to static quant

Test Plan:
python test/test_quantize_script.py

Sample output for single layer FC

.graph(%self : __torch__.___torch_mangle_4.M,
      %x.2 : Tensor):
  %_observer_1 : __torch__.torch.quantization.observer.MinMaxObserver = prim::GetAttr[name="_observer_1"](%self)
  %x.1 : Tensor = prim::CallMethod[name="forward"](%_observer_1, %x.2)
  %2 : __torch__.torch.nn.modules.linear.___torch_mangle_5.Linear = prim::GetAttr[name="fc"](%self)
  %3 : Tensor = prim::CallMethod[name="forward"](%2, %x.1) # test/test_quantize_script.py:19:23
  return (%3)

graph(%self : __torch__.torch.nn.modules.linear.___torch_mangle_5.Linear,
      %input.1 : Tensor):
 %2 : Function = prim::Constant[name="linear"]()
 %3 : Tensor = prim::GetAttr[name="weight"](%self)
 %_observer_0 : __torch__.torch.quantization.observer.MinMaxObserver = prim::GetAttr[name="_observer_0"](%self)
 %7 : Tensor = prim::CallMethod[name="forward"](%_observer_0, %3)
 %4 : Tensor = prim::GetAttr[name="bias"](%self)
 %5 : Tensor = prim::CallFunction(%2, %input.1, %7, %4) # /home/supriyar/miniconda3/envs/pytorch_py3/lib/python3.7/site-packages/torch/nn/modules/linear.py:87:15
 return (%5)

Imported from OSS

Differential Revision: D20599144

fbshipit-source-id: 9a8fa0e8655b9908826b981dce8a11d86efce5df

nikitaved committed:
simpler 'cpu_scatter_gather_base_kernel' (#34690)

Summary:
Simplifies `cpu_scatter_gather_base_kernel` to accept only binary operations and spares them from doing redundant checks.
CC v0dro
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34690

Differential Revision: D20604814

Pulled By: ngimel

fbshipit-source-id: 5e22c2f39a8e2861dc763454c88796d1aa38d2eb

malfet committed:
Disable C4251 when compiling cpp_extensions on Windows (#35272)

Summary:
Otherwise, VC++ will warn that every exposed C++ symbol, for example:
```
include\c10/core/impl/LocalDispatchKeySet.h(53): warning C4251: 'c10::impl::LocalDispatchKeySet::included_': class 'c10::DispatchKeySet' needs to have dll-interface to be used by clients of struct 'c10::impl::LocalDispatchKeySet'
```
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35272

Test Plan: CI

Differential Revision: D20623005

Pulled By: malfet

fbshipit-source-id: b635b674159bb9654e4e1a1af4394c4f36fe35bd

ezyang committed:
Trying pinning pyyaml and setuptools on macos to older version (#35296)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35296

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D20624843

Pulled By: ezyang

fbshipit-source-id: 9028f1dd62d0c25e916eb4927fd8dd6acbd88886

XiaobingSuper committed:
remove cadd_avx2 dead code (#34883)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/34883

Test Plan: Imported from OSS

Differential Revision: D20611526

Pulled By: ngimel

fbshipit-source-id: 78c80b7361119fc8d2b9f6b4f0c86b61723fe05d

XiaobingSuper committed:
Optimize min and max(reduce_dim) performance on CPU (#34875)

Summary:
This PR is about improve min and max(reduce_dim) performance on CPU.
Test script:
```
 import torch
 import torch.nn as nn
 import time

 torch.manual_seed(0)

 def _time():
     return time.time()

 device = "cpu"
 torch.set_num_threads(1)

 #warm up
 for n in [10, 200]:
     # contiguous
     # input = torch.randn(n, n, n, requires_grad=False, device=device)
     # discontiguous
     input = torch.randn(n, 2*n, n, requires_grad=False, device=device)[:, :n, :]
     for dim in range(input.dim()):
         for i in range(1000):
             output = input.min(dim)
             #output = input.max(dim)

 for n in [10, 200]:
     # contiguous input
     # input = torch.randn(n, n, n, requires_grad=False, device=device)
     # discontiguous
     input = torch.randn(n, 2*n, n, requires_grad=False, device=device)[:, :n, :]
     for dim in range(input.dim()):
         fwd_t = 0
         for i in range(10000):
             t1 = _time()
             output = input.min(dim)
             #output = input.max(dim)
             t2 = _time()
             fwd_t = fwd_t + (t2 -t1)
         fwd_avg = fwd_t / 10000 * 1000
         print("size = (%d, %d, %d); reduce dim=%d; compute time is %.4f(ms)" % (n, n, n, dim, fwd_avg))
```
Test device: **skx-8180**.

### Contiguous case.
- num_threads = 56

  | Bef(ms) | Bef(ms) | Bef(ms) | Aft(ms) | Aft(ms) | Aft(ms)
-- | -- | -- | -- | -- | -- | --
size | dim=0 | dim=1 | dim=2 | dim=0 | dim=1 | dim=2
n=10 | 0.0243 | 0.0243 | 0.0244 | 0.0063 | 0.0065 | 0.0063
n=200 | 0.9615 | 0.9453 | 0.7772 | 0.2937 | 0.2675 | 0.2607

- num_threads = 1

  | Bef(ms) | Bef(ms) | Bef(ms) | Aft(ms) | Aft(ms) | Aft(ms)
-- | -- | -- | -- | -- | -- | --
size | dim=0 | dim=1 | dim=2 | dim=0 | dim=1 | dim=2
n=10 | 0.0126 | 0.0126 | 0.0114 | 0.0062 | 0.0065 | 0.0064
n=200 | 32.1276 | 33.3489 | 29.0757 | 8.0556 | 7.0188 | 6.5014

### Discontiguous case.
- num_threads = 56

  | Bef(ms) | Bef(ms) | Bef(ms) | Aft(ms) | Aft(ms) | Aft(ms)
-- | -- | -- | -- | -- | -- | --
size | dim=0 | dim=1 | dim=2 | dim=0 | dim=1 | dim=2
n=10 | 0.0106 | 0.0115 | 0.0131 | 0.0063 | 0.0066 | 0.0065
n=200 | 14.652 | 15.3496 | 9.8153 | 0.2946 | 0.2708 | 0.267
- num_threads = 1

  | Bef(ms) | Bef(ms) | Bef(ms) | Aft(ms) | Aft(ms) | Aft(ms)
-- | -- | -- | -- | -- | -- | --
size | dim=0 | dim=1 | dim=2 | dim=0 | dim=1 | dim=2
n=10 | 0.0108 | 0.0116 | 0.0132 | 0.0058 | 0.0062 | 0.0061
n=200 | 12.5132 | 13.0785 | 9.6738 | 8.3733 | 7.3051 | 6.4566

https://github.com/pytorch/pytorch/issues/24671 and https://github.com/pytorch/pytorch/issues/24672 are also fixed.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34875

Differential Revision: D20605596

Pulled By: ngimel

fbshipit-source-id: 08fd4dacd1db63309123d7ec5942a4b8a0071896

Commit:
Update XNNPACK to Git revision 1b354636b5942826547055252f3b359b54acff95. (#35081)

Summary:
Required to fix a build issue in https://github.com/pytorch/pytorch/issues/33766.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35081

Reviewed By: dreiss

Differential Revision: D20567230

Pulled By: AshkanAliabadi

fbshipit-source-id: 1ed61708851402f60b80abc818ae7330e43adb83

zasdfgbnm committed:
Make GPU loops support mutable lambda (#35015)

Summary:
I will need it for https://github.com/pytorch/pytorch/pull/34004

The `mutable` qualifier allows a lambda to capture some values, and modify its own copy. This would be useful for random kernels: we capture a `state` of RNG, initialize it when it first run, and the initialized stated will be used later:

```C++
gpu_kernel(iter, [state, initialized](scalar_t arg) mutable -> scalar_t {
  if (!initialized) {
    curand_init(..., state);
    initialized = true;
  }
  return some_math(curand_uniform(state), arg);
}
```

The `operator()` of `mutable` lambda is not `const`, so we can not pass it as constant reference. It can not be called inside a non-`mutable` lambda either.

Example usage:

```C++
auto t = at::empty({4096}, kCUDA);
float thread_work_index_ = 0;
auto iter = TensorIterator::nullary_op(t);
gpu_kernel(iter, [thread_work_index_]GPU_LAMBDA() mutable -> float {
  return thread_work_index_++;
});
```
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35015

Differential Revision: D20624698

Pulled By: ngimel

fbshipit-source-id: 06e3987793451cd514181d20252510297e2d28a9

r-zenine committed:
Add's workaround for ScalarType::Byte for cuda (#35027)

Summary:
This PR add's a workaround for `cuda` for `ScalarType::Byte` for the `AT_DISPATCH_*` macros.
As discussed here:
https://github.com/pytorch/pytorch/issues/34826
Pull Request resolved: https://github.com/pytorch/pytorch/pull/35027

Differential Revision: D20596555

Pulled By: colesbury

fbshipit-source-id: 72e842603723a5aa146e4224e79befafc62f2624

Commit:
[jit] Add lazy script decorator (#34935)

Summary:
Stacked PRs
 * #34938 - [jit] Remove stray `script`
 * **#34935 - [jit] Add lazy script decorator**

Some users maintain libraries of code that is largely trace-able but not
script-able. However, some functions may need to be `torch.jit.script`ed if
they contain control flow so the tracer will use the compiler version.
This however impacts library start up time as in #33418, so this PR adds
a workaround in the form of a `torch.jit._lazy_script_while_tracing`
that will only initialize the compiler if the function is called while
actually tracing.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/34935

Pulled By: driazati

Differential Revision: D20569778

fbshipit-source-id: d87c88c02b1abc86b283729ab8db94285d7d4853

Commit:
[jit] Remove stray `@script` (#34938)

Summary:
Stacked PRs
 * **#34938 - [jit] Remove stray `script`**
 * #34935 - [jit] Add lazy script decorator

Pull Request resolved: https://github.com/pytorch/pytorch/pull/34938

Pulled By: driazati

Differential Revision: D20569793

fbshipit-source-id: 1f126646f7bd7c4ea972e15023eaa60f0e301351

jerryzh168 committed:
[quant][graphmode] Add quantization support for aten::cat (#34346)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/34346

Test Plan:
python test/test_jit.py

Imported from OSS

Differential Revision: D20589145

fbshipit-source-id: c9159fffa88cf25fcdccfcc4eef2622cf4b250b5

Commit:
Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py (#35189)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/35189

Test Plan: Imported from OSS

Differential Revision: D20588197

Pulled By: yf225

fbshipit-source-id: 5a28159b653895678c250cbc0c1ddd51bc7a3123

Commit:
Merge branch 'master' of https://github.com/protonu/pytorch

Commit:
Merge branch 'master' of https://github.com/protonu/pytorch
